cfdm/data/subarray/mixin/pointtopology.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+from ....functions import integer_dtype
+
+
+class PointTopology:
+    """Mixin class for point topology array compressed by UGRID.
+
+    Subclasses must also inherit from `MeshSubarray`.
+
+    .. versionadded:: (cfdm) UGRIDVER
+
+    """
+
+    def __getitem__(self, indices):
+        """Return a subspace of the uncompressed data.
+
+        x.__getitem__(indices) <==> x[indices]
+
+        .. versionadded:: (cfdm) UGRIDVER
+
+        """
+        from math import isnan
+
+        from scipy.sparse import csr_array
+
+        start_index = self.start_index
+        node_connectivity = self._select_data(check_mask=False)
+
+        masked = np.ma.isMA(node_connectivity)
+
+        largest_node_id = node_connectivity.max()
+        if not start_index:
+            # Add 1 to remove all zeros (0 is the fill value in the
+            # sparse array), first making sure that the datatpe can
+            # handle it.
+            if largest_node_id == np.iinfo(node_connectivity.dtype).max:
+                node_connectivity = node_connectivity.astype(int, copy=False)
+
+            node_connectivity = node_connectivity + 1
+            largest_node_id += 1
+
+        p = 0
+        pointers = [0]
+        cols = []
+        u = []
+
+        pointers_append = pointers.append
+        cols_extend = cols.extend
+        u_extend = u.extend
+
+        # WARNING: This loop is a potential performance bottleneck
```

Comment:
```suggestion
        # WARNING (TODO): This loop is a potential performance bottleneck.
```

or similar perhaps, to add a 'TODO' here since it would be good to revisit in case there is a more performant way?